### DISCUSSION TITLE
SSL error when opening the socket

### DISCUSSION THREAD

**Discussion by vishalkc9565** [deepgram/community#560] [General help] [answered]
### Which Deepgram product are you using?

Deepgram API

### Details

"""
https://developers.deepgram.com/docs/getting-started-with-live-streaming-audio
"""

from dotenv import load_dotenv

from deepgram import (
    DeepgramClient,
    DeepgramClientOptions,
    LiveTranscriptionEvents,
    LiveOptions,
    Microphone,
)

load_dotenv()

def main():
    try:
        # Create a Deepgram client using the API key from environment variables
        # (ie export DEEPGRAM_API_KEY="YOUR_API_KEY")
        deepgram = DeepgramClient(api_key="Xxxxxxx")

        dg_connection = deepgram.listen.live.v("1")

        def on_message(self, result, **kwargs):
            sentence = result.channel.alternatives[0].transcript
            if len(sentence) == 0:
                return
            print(f"transcription: {sentence}")
            
        def on_error(self, error, **kwargs):
            print(f"\n\n{error}\n\n")

        dg_connection.on(LiveTranscriptionEvents.Transcript, on_message)
        dg_connection.on(LiveTranscriptionEvents.Error, on_error)

        options = LiveOptions(
            smart_format=True,
            language="en-US",
            # encoding="linear16",
            channels=1,
            sample_rate=16000,
        )
        dg_connection.start(options,ssl=False)

        # Open a microphone stream
        microphone = Microphone(dg_connection.send)

        # start microphone
        microphone.start()

        # wait until finished
        input("Press Enter to stop recording...\n\n")

        # Wait for the microphone to close
        microphone.finish()

        # Indicate that we've finished
        dg_connection.finish()

        print("Finished")

    except Exception as e:
        print(f"Could not open socket: {e}")
        return

if __name__ == "__main__":
    main()

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by davidvonthenen** [comment]
@vishalkc9565 can you provide more details about the error?

Perhaps turn on debug logging?
https://github.com/deepgram/deepgram-python-sdk/blob/main/examples/streaming/http/main.py#L26-L29

**Reply by vishalkc9565** [reply]
I am running your code as it is as given at https://github.com/deepgram/deepgram-python-sdk/blob/main/examples/streaming/microphone/main.py

Have you run the code on local machine.
![image](https://github.com/deepgram/community/assets/146014256/7ec0779e-f08d-43cd-a9d7-f07924eb93b4)

Not able to figure out this

**Comment by davidvonthenen** [comment]
Are you on MacOS?

**Reply by vishalkc9565** [reply]
yes

**Reply by davidvonthenen** [reply, marked as answer]
On MacOS, just browse to Applications/Python 3.X and double-click `Install Certificates.command`.

**Reply by vishalkc9565** [reply]
ok, thanks a lot. It solved the issue
